src/main/java/com/learning/browsers/BrowserDriverGenerator.java

```diff
@@ -0,0 +1,30 @@
+package com.learning.browsers;
+
+import com.learning.configuration.PropertiesReader;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+
+public class BrowserDriverGenerator implements BrowserDriver {
+
+  private static final String BROWSER_PROPERTIES_NAME = "browser";
+  private static final String PATH_TO_EDGE_DRIVER = "pathToEdgeDriver";
+  private final PropertiesReader propertiesReader;
+
+  public BrowserDriverGenerator() {
+    propertiesReader = new PropertiesReader();
```

Comment:
Instead of reading properties each time they are used, they could also be turned into singleton, and injected with Guice.

Reply:
Wait... I see you actually have Singleton annotation on `PropertiesReader` you just don't use it that way. I think you should :D

Reply:
I overlooked this one.